src/cfnlint/rules/resources/ecs/FargateDeploymentSchedulingStrategy.py

```diff
@@ -0,0 +1,28 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+from cfnlint.rules import CloudFormationLintRule, RuleMatch
+
+
+class FargateDeploymentSchedulingStrategy(CloudFormationLintRule):
+    id = "E3044"
+    shortdesc = "Check Fargate service scheduling strategy"
+    description = "Check that Fargate service scheduling strategy is REPLICA"
+    source_url = "https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ecs-service.html#cfn-ecs-service-schedulingstrategy"
+    tags = ["properties", "ecs", "service", "container", "fargate"]
+
+    def match(self, cfn):
+        matches = []
+        ecs_services = cfn.get_resource_properties(["AWS::ECS::Service"])
+
+        for ecs_service in ecs_services:
+            path = ecs_service["Path"]
+            properties = ecs_service["Value"]
+            if isinstance(properties, dict):
+                if properties.get("LaunchType", None) != "Fargate":
+                    continue
+                if properties.get("SchedulingStrategy", None) != "REPLICA":
```

Comment:
I think this would trigger a few false positives. We should probably validate that SchedulingStrategy is a string for the `!=`

```yaml
Parameters:
  SchedulingStrategy:
    Type: String
  Environment:
    Type: String
Conditions:
  IsProduction: !Equals [!Ref Environment, 'Prod']
Resources:
  Service1:
    Type: AWS::ECS::Service
    Properties:
      LaunchType: Fargate
      SchedulingStrategy: !Ref SchedulingStrategy
  Service2:
    Type: AWS::ECS::Service
    Properties:
      LaunchType: EC2
      SchedulingStrategy: !If [IsProduction, DAEMON, REPLICA]
```